auto_landing/src/tracking.cpp

```diff
@@ -0,0 +1,93 @@
+#include<auto_landing/tracking.hpp>
+
+namespace ariitk::auto_landing {
+
+Tracking::Tracking() :
+    height_(1) {}
+
+void Tracking::init(ros::NodeHandle& nh, char** argv) {
+    char* end_ptr;
+    husky_relative_x_ = strtod(argv[2], &end_ptr);
+    husky_relative_y_ = strtod(argv[4], &end_ptr);
+
+    ROS_INFO ("Husky spawned at %lf %lf", husky_relative_x_, husky_relative_y_);
+
+    set_firefly_pose_ = nh.advertise<geometry_msgs::PoseStamped>("commad_pose", 1);
+    quad_pose_ = nh.subscribe("quad_odometry", 1, &Tracking::quadPoseCallback, this);
+    gazebo_model_state_=nh.subscribe("model_state",1,&Tracking::modelStateCallback,this);
+    landing_client_ = nh.serviceClient<std_srvs::Trigger>("to_land");
+}
+
+void Tracking::run(){
+    landing_service_.response.success=false;
+    
+    if((abs(husky_odom_.pose.pose.position.x-quad_odom_.pose.pose.position.x) < 0.1) && (abs(husky_odom_.pose.pose.position.x-quad_odom_.pose.pose.position.x) < 0.1)) {
+        ROS_INFO("Over Husky.");
+        landing_client_.call(landing_service_);
+    }
+    
+    if((abs(husky_cmd_vel_[0].linear.x)>0.0001 || abs(husky_cmd_vel_[0].linear.y)>0.0001)){
+        quadSetPointUpdate();
+    }
+    
+    set_firefly_pose_.publish(setpt_);
+}
+
+void Tracking::modelStateCallback(const gazebo_msgs::ModelStates& msg){
+    
+    time_[0]= time_[1];
```

Comment:
you don't need to do this. the subscriber callback rate is equal to the topic publish rate, so t2-t1 will always be 1/rate. you can confirm this by printing them. so remove these lines and the variables, and use ```inv_state_publish_rate_``` instead for t2-t1